### ISSUE TITLE
Add "You have been invited as an author" popup

### ISSUE THREAD

**Issue by 5e-Cleric** [naturalcrit/homebrewery#3945]
In the editor page, right now, a user can be added as an `invited author`, which means said user has access to the edit page, but is not an author per-se. To promote themselves to an actual author, the user must save an edit to the file. 

Right now, users usually get confused around this feature, because it is barely explained, i propose we add two features:
1. a explain icon besides the feature in the metadata panel, like so:

![image](https://github.com/user-attachments/assets/53e308d0-eec7-41b8-8a67-fc8e1b461e59)
 With an explanation on how the feature works.
 
2. A popup in the edit page for when a user enters the edit page of the brew, while being in the invited authors list, to explain the feature to them, giving them clear instructions about how to become or not become an author, as well as to how to remove themselves as author.

3. Change the deletion button text from deleting brew to `remove self from author list` if the user is an invited author, or an author but not the only author.

**Comment by G-Ambatte** [comment]
It might also be useful to pop a notification in the Share Page if the current logged in user is an invited author, probably with a link to the Edit Page.

**Comment by ericscheid** [comment]
A reminder that one of the design goals of the invited author functionality was that it should not be available to be abused as a channel or vector for abuse or harassment.

Note though that none of the suggestions here do that — but they are getting close. Take this as only a caution regarding any additional messaging options.

**Comment by G-Ambatte** [comment]
There was a recent Discord user who reported that they had been invited to co-author a brew, but had no idea how to access it - the original author sent them the Share link, not the Edit link, and currently there is nothing in place to direct an Invited Author to the correct page.

**Comment by calculuschild** [comment]
How about this: instead of a popup or messaging system, we just add a new section to the user page with author invitations?

##### *Author Invitations*

##### User's Published Brews

##### User's Unpublished Brews

**Comment by ericscheid** [comment]
> How about this: instead of a popup or messaging system, we just add a new section to the user page with author invitations?
> 
> Author Invitations
> 
> User's Published Brews
> 
> User's Unpublished Brews
> 
100% no. That is a vector for harassment. Imagine being on the receiving end of a hate campaign, with your personal home page overwhelmed with brews with utterly noxious titles and descriptions.

**Comment by G-Ambatte** [comment]
> 100% no.

I think there's a number of steps to put in place first. 

- publish acceptable community content standard (including penalties for failing to meet them)
- create a reporting system (brews & user)
- improve/finish the locking system

I should also mention, especially in the light of recent user reports, that I believe we're almost certainly going to have to take these steps anyway, regardless of whether "Invited Brews" ever appears in the User Page.

**Comment by 5e-Cleric** [comment]
Well, i do prefer my original idea instead of the userpage.

**Comment by 5e-Cleric** [comment]
Well, i do prefer my original idea instead of the userpage's.

**Comment by calculuschild** [comment]
I'm seeing multiple related, but different issues here. This issue focuses on #1 and #2:

1) **Users don't know how to initiate an invite of co-authors** (could be solved by better documentation / UI, as proposed by 5e-Cleric)
2) **Users don't know how to transition from invited to a full author** (5e-Cleric proposes a notification with instructions upon opening the edit link)
3) **Users don't know how to access an invited brew** (either some kind of notification to the invited author, or better instruction for the owning author)
4) **Users don't know how to give their co-authors access** (probably solved if #3 is solved)

Its ok if this issue focuses on #1 and #2.

The Userpage suggestion was in response to "discord user didn't know how to access the brew". I apologize that was going off topic into the #3 and #4 issues.

> 100% no

It would seem that *any* type of signal to the invited author for #3 and #4 is a vector for harassment. This means the only way forward for those is better instructions on the owner end.

**Comment by G-Ambatte** [comment]
I was moments away from posting a very similar response with a very similar list. Agreed, we were getting into some larger issues there.

> discord user didn't know how to access the brew

They had been sent a share link, not an edit - so better instructions for the initiator on how the system works might have solved that one, too.

But that was also my basis for suggesting a Share Page pop up along the lines of: "Hey, you've been invited to collaborate on this brew! Click here to start that now!"
My theory being that no vector for harassment is introduced, as the user is already viewing the material in question.

![image](https://github.com/user-attachments/assets/cd24b8b9-2652-4bc3-9b84-1aeee963d191)

I have GOT to stop finding potential uses for Dicey.

**Comment by dbolack-ab** [comment]
I think it would be reasonable if a *logged-in* user is prompted when they are an invited user on a shareID or editID path.

The prompt should allow them to accept, decline, or sleep the invite. If they accept or decline, the brew should be updated. If they sleep, session data should be stored so that the prompt can query and not pop up for a week.

I don't think we need to provide instructions - just a simple, actionable interface along with a "not right now!" option.

**Comment by calculuschild** [comment]
The concern is if anyone happens to know your username and wants to be abusive, they can just set continually invite you to documents you don't want. Spam notifications.

So the preferred approach is to not let users trigger notifications on another user like this at all.

**Comment by 5e-Cleric** [comment]
> I think it would be reasonable if a _logged-in_ user is prompted when they are an invited user on a shareID or editID path.

Not logged in users just cannot be invited so ... yes

> The prompt should allow them to accept, decline, or sleep the invite. If they accept or decline, the brew should be updated. If they sleep, session data should be stored so that the prompt can query and not pop up for a week.
> 
> I don't think we need to provide instructions - just a simple, actionable interface along with a "not right now!" option.

I agree 100%


> The concern is if anyone happens to know your username and wants to be abusive, they can just set continually invite you to documents you don't want. Spam notifications.
> 
> So the preferred approach is to not let users trigger notifications on another user like this at all.

Abquintic specified to show a popup *when* the invited author enters the share or edit pages. no spam that way.

**Comment by dbolack-ab** [comment]
> > > The concern is if anyone happens to know your username and wants to be abusive, they can just set continually invite you to documents you don't want. Spam notifications. So the preferred approach is to not let users trigger notifications on another user like this at all.
> 
> > Abquintic specified to show a popup when the invited author enters the share or edit pages. no spam that way.
> 
> Defnitely no notifications outwith the page being invited to.
> 

Then you have a (*&)*( useless feature.

**Comment by ericscheid** [comment]
> > Defnitely no notifications outwith the page being invited to.
> 
> Then you have a (_&)_( useless feature.

I'm agreeing with Abquintic that a notification can be shown on the share or edit pages, and not be shown elsewhere. Don't show it on random other pages, not shown on their userpage, not shown on the home page .. but do show it on the page in question. How is that useless?

**Comment by calculuschild** [comment]
I agree if we are talking about notifications *on the invited page itself*. That wasn't clear to me when I first read that. I only saw the "when a user is logged in" part.

**Comment by 5e-Cleric** [comment]
I forgot this issue existed when i created #4101

**Comment by dbolack-ab** [comment]
We should put this on the V4 release roadmap. This is a quality-of-life need.

**Comment by 5e-Cleric** [comment]
This can be done before that, not sure if we need to wait.

**Comment by dbolack-ab** [comment]
This should be considered dependent on https://github.com/naturalcrit/homebrewery/pull/4122 or replacement work.
